modules/phase_field/src/materials/DeformedGrainMaterial.C

```diff
@@ -0,0 +1,104 @@
+  params.addParam<Real>("GBMobility", 2.0e-13, "GB mobility input in m^4/(J*s)");
+  params.addParam<Real>("GBE", 1.0, "Grain boundary energy in J/m^2");
+  params.addParam<Real>("Disloc_Den", 9.0e15, "Dislocation Density in m^-2");
+  params.addParam<Real>("Elas_Mod", 2.50e10, "Elastic Modulus in J/m^3");
+  params.addParam<Real>("Burg_vec", 3.0e-10, "Length of Burger Vector in m");
+  params.addRequiredParam<UserObjectName>("grain_tracker", "The GrainTracker UserObject to get values from.");
+  return params;
+}
+
+DeformedGrainMaterial::DeformedGrainMaterial(const InputParameters & parameters) :
+    Material(parameters),
+    _length_scale(getParam<Real>("length_scale")),
+    _int_width(getParam<Real>("int_width")),
+    _time_scale(getParam<Real>("time_scale")),
+    _GBMobility(getParam<Real>("GBMobility")),
+    _GBE(getParam<Real>("GBE")),
+    _Disloc_Den(getParam<Real>("Disloc_Den")),
+    _Elas_Mod(getParam<Real>("Elas_Mod")),
+    _Burg_vec(getParam<Real>("Burg_vec")),
+    _kappa(declareProperty<Real>("kappa_op")),
+    _gamma(declareProperty<Real>("gamma_asymm")),
+    _L(declareProperty<Real>("L")),
+    _mu(declareProperty<Real>("mu")),
+    _tgrad_corr_mult(declareProperty<Real>("tgrad_corr_mult")),
+    _beta(declareProperty<Real>("beta")),
+    _Disloc_Den_i(declareProperty<Real>("Disloc_Den_i")),
+    _rho_eff(declareProperty<Real>("rho_eff")),
+    _Def_Eng(declareProperty<Real>("Def_Eng")),
+    _ndef(getParam<unsigned int>("ndef")),
+    _op_num(coupledComponents("v")),
+    _grain_tracker(getUserObject<GrainTrackerInterface>("grain_tracker")),
+    _kb(8.617343e-5), //Boltzmann constant in eV/K
+    _JtoeV(6.24150974e18) // Joule to eV conversion
+{
+  if (_op_num == 0)
+    mooseError("Model requires op_num > 0");
+
+  _vals.resize(_op_num);
+  for (unsigned int i=0; i < _op_num; ++i)
+    _vals[i] = &coupledValue("v", i);
+}
+
+void
+DeformedGrainMaterial::computeQpProperties()
+{
+  _Disloc_Den_i[_qp] = _Disloc_Den * (_length_scale * _length_scale);
+  Real rho_i = _Disloc_Den_i[_qp];
+  Real rho0 = 0.0;
+  Real SumEtai2 = 0.0;
+  for (unsigned int i = 0; i < _op_num; ++i)
+    SumEtai2 += (*_vals[i])[_qp]*(*_vals[i])[_qp];
+
+  // calculate effective dislocation density and assign zero dislocation densities to undeformed grains
+  const std::vector<std::pair<unsigned int, unsigned int> > & active_ops = _grain_tracker.getElementalValues(_current_elem->id());
+  unsigned int n_active_ops= active_ops.size();
+  if (n_active_ops < 1 && _t_step > 0)
+    mooseError("No active order parameters");
+
+  // loop over active OPs
+  for (unsigned int op = 0; op < n_active_ops; ++op)
+  {
+    // First position of the active ops contains grain number
+    unsigned int grn_index = active_ops[op].first;
+
+    // Second position contains the order parameter index
+    unsigned int op_index = active_ops[op].second;
+    if (grn_index >= _ndef)
+      rho_i = 0.0;
```

Comment:
`rho_i ` will stay `0.0` even if the next entry in the list of op/grain pairs has a lower `grn_index` (which by the way I'd like to see renamed to `grain_index` - we should spend the two extra letters to make this more readable!).

This needs an 
```
else 
  rho_i = _Disloc_Den_i[_qp];
```

(and remove the ` = _Disloc_Den_i[_qp]` at the declaration of `rho_i`)